enumerable.rb

```diff
@@ -0,0 +1,109 @@
+module Enumerable
+  def my_each
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    each do |i|
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    (0..size - 1).my_each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    # an empty array to store our selected element
+    arr = []
+    (0...size - 1).my_each do |i|
+      arr << self[i] if yield(self[i])
+    end
+    arr
+  end
+
+  def my_all?
+    status = true
+    my_each do |x|
+      status = false if x.nil? || x == false
+    end
+    return status unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 unless yield self[i]
+    end
+    count.zero?
+  end
+
+  def my_any?
+    status = true
+    my_each do |x|
+      status = false if x.nil? || x == false
+    end
+    return status unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 if yield self[i]
+    end
+    count != 0
+  end
```

Comment:
#### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

     > this [nil, false, true, []].my_any? == [nil, false, true, []].any? => true

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

     > this [1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

     > %w[dog door rod blade].my_any?(/z/) == %w[dog door rod blade].any?(/z/) => false

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

     > [3,4,7,11].my_any?(3) == [3,4,7,11].any?(3) => true